extensions/ql-vscode/src/databases/local-databases-ui.ts

```diff
@@ -827,7 +827,7 @@ export class DatabaseUI extends DisposableObject {
     return this.databaseManager.currentDatabaseItem;
   }
 
-  private async promptForDatabase(): Promise<void> {
+  public async promptForDatabase(): Promise<void> {
```

Comment:
I'm not particularly happy about making all these methods public... Is this okay, or is there a better way?

Reply:
A way to do that would be to test the public function that is using this one (e.g. `downloadDatabaseFromGitHub`). That's the public interface of the module, and what callers will use, so you would be testing the same flows as the callers. IMO that would be the best way to go than making 3 members public. Would that make the test too difficult/complex?

Another approach is to move `promptForDatabase` in its own file/module, showing that in itself it has a public interface that it can be used and tested independently.  

Having said all that, I want to call out that there are often cases where we have to make things public for testing and I think that's acceptable and better than not having tests. Ideally we'd add a code comment to tell people that a member is only exported for testing and should not be used in non test code.

Reply:
> A way to do that would be to test the public function that is using this one (e.g. `downloadDatabaseFromGitHub`). That's the public interface of the module, and what callers will use, so you would be testing the same flows as the callers. IMO that would be the best way to go than making 3 members public. Would that make the test too difficult/complex?

`promptForDatabase` is only called from `getDatabaseItemInternal` (which is also private). I could try testing the public wrapper `getDatabaseItem` though!

Reply:
Ah sorry, it looks like there's a `promptForDatabases` (plural) and I was looking at that instead 🤦‍♀️ 

Thanks, it would be good to try against `getDatabaseItem`. But as I mentioned, if it's too hard it may not be worth the trouble.